src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -330,6 +409,242 @@ public void jobFinished () {
         NotifyUsersForSubscriptionJob.createNotification("deployment-updated", deployment.id, message);
     }
 
+    public void replaceEC2Servers() {
+        try {
+            // First start graph-building instance and wait for graph to successfully build.
+            List<Instance> instances = startEC2Instances(1);
+            if (instances.size() > 1) {
+                // FIXME is this check/shutdown entirely unnecessary?
+                status.fail("CRITICAL: More than one server initialized for graph building. Cancelling job. Please contact system administrator.");
+                // Terminate new instances.
+                // FIXME Should this ec2 client be longlived?
+                ec2.terminateInstances(new TerminateInstancesRequest(getIds(instances)));
+            }
+            // FIXME What if instances list is empty?
+            MonitorServerStatusJob monitorInitialServerJob = new MonitorServerStatusJob(owner, deployment, instances.get(0), otpServer);
+            monitorInitialServerJob.run();
+            status.update("Graph build is complete!", 50);
+            // Spin up remaining servers which will download the graph from S3.
+            int remainingServerCount = otpServer.instanceCount <= 0 ? 0 : otpServer.instanceCount - 1;
+            if (remainingServerCount > 0) {
+                // Spin up remaining EC2 instances.
+                List<Instance> remainingInstances = startEC2Instances(remainingServerCount);
+                instances.addAll(remainingInstances);
+                // Create new thread pool to monitor server setup so that the servers are monitored in parallel.
+                ExecutorService service = Executors.newFixedThreadPool(remainingServerCount);
+                for (Instance instance : remainingInstances) {
+                    // Note: new instances are added
+                    MonitorServerStatusJob monitorServerStatusJob = new MonitorServerStatusJob(owner, deployment, instance, otpServer);
+                    service.submit(monitorServerStatusJob);
+                }
+                // Shutdown thread pool once the jobs are completed and wait for its termination. Once terminated, we can
+                // consider the servers up and running (or they have failed to initialize properly).
+                service.shutdown();
+                service.awaitTermination(4, TimeUnit.HOURS);
+            }
+            String finalMessage = "Server setup is complete!";
+            if (otpServer.instanceIds != null) {
```

Comment:
This looks like the default behavior for deployment involves decommissioning other ec2 instances. This wasn't written in the tutorial and I think it's important to note in case someone is just trying things out and overwrites an existing deployment. The UI ought to have some kind of check or warning describing how many instances are currently running.